WebApp/src/main/java/com/isa/webapp/controller/TeacherController.java

```diff
@@ -1,115 +1,80 @@
+import com.isa.webapp.model.*;
+import com.isa.webapp.repository.GradeRepository;
+import com.isa.webapp.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
 import org.springframework.security.core.annotation.AuthenticationPrincipal;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.*;
 
 import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.ArrayList;
+import java.time.LocalDate;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 
 @Controller
+@RequiredArgsConstructor
 public class TeacherController {
 
-    private final StudentService studentService;
+    private final UserRepository userRepository;
 
-    @Autowired
-    public TeacherController(StudentService studentService) {
-        this.studentService = studentService;
-    }
-
-    private static final String USERS_JSON_FILE = "users.json";
+    private final GradeRepository gradeRepository;
 
     @GetMapping("/teacher/students")
     public String showStudentList(Model model, @AuthenticationPrincipal UserDetails userDetails) throws IOException {
-        boolean isTeacher = userDetails.getAuthorities().stream()
-                .anyMatch(a -> a.getAuthority().equals("TEACHER"));
+        boolean isTeacher = userDetails.getAuthorities().stream().anyMatch(a -> a.getAuthority().equals("TEACHER"));
         User user = (User) userDetails;
-        List<User> students = getStudentsForTeacher();
+        /*        List<User> students = getStudentsForTeacher();*/
+        List<User> students = userRepository.findAllByUserRole(UserRole.STUDENT);
         model.addAttribute("students", students);
         model.addAttribute("isTeacher", isTeacher);
         model.addAttribute("username", user.getFirstName() + " " + user.getLastName());
         return "teacher_student_list";
     }
 
-    @GetMapping("/teacher/add-grade/{studentId}")
-    public String showAddGradeForm(@PathVariable String studentId,
-                                   Model model,
-                                   @AuthenticationPrincipal UserDetails userDetails) {
-        boolean isTeacher = userDetails.getAuthorities().stream()
-                .anyMatch(a -> a.getAuthority().equals("TEACHER"));
+    @GetMapping("/teacher/add-grade/{studentUuid}")
+    public String showAddGradeForm(@PathVariable String studentUuid, Model model, @AuthenticationPrincipal UserDetails userDetails) {
+        boolean isTeacher = userDetails.getAuthorities().stream().anyMatch(a -> a.getAuthority().equals("TEACHER"));
```

Comment:
to też do funkcji

Reply:
Nie do końca zrozumiałem, co mam "wrzucić" do funkcji..

Reply:
napisałem Ci wyżej przykład z tym `isTeacher`